### ISSUE TITLE
Container Source does not get annotation for outbound network

### ISSUE THREAD

**Issue by sebgoa** [knative/eventing-sources#223]
## Expected Behavior

A pod created by a containersource should have the annotation for Istio to let the network out.
If you follow this setup:

https://github.com/knative/docs/blob/master/serving/outbound-network-access.md

A pod should have the following type of annotation:

```
traffic.sidecar.istio.io/includeOutboundIPRanges: 10.28.0.0/14
```

## Actual Behavior

Pod created by container source does not get the "traffic sidecar" annotation and therefore traffic from the source cannot egress.

## Steps to Reproduce the Problem

1. Get a cluster on GKE
2. Install 0.3.0 eventing
3. Install dynamodb source from https://github.com/triggermesh/knative-lambda-sources
4. see that the source cannot reach dynamodb servers.

**Comment by sebgoa** [comment]
for reference: networking function in serving:
https://github.com/knative/serving/blob/master/pkg/network/network.go

Loaded in:
https://github.com/knative/serving/blob/master/pkg/reconciler/v1alpha1/revision/config/store.go

**Comment by evankanderson** [comment]
Should this be sidecar annotations or [Istio ServiceEntries](https://istio.io/docs/reference/config/istio.networking.v1alpha3/#ServiceEntry)?

**Comment by sebgoa** [comment]
You can add the sidecar annotation to the container source deployment which fixes the issue, but it will get removed after a while (reconciliation I suppose).

It seems with Istio 1.1 default will be open so we won't need the annotation anymore.

**Comment by vaikas-google** [comment]
Fixed in #285 
https://github.com/knative/eventing-sources/pull/285

This allows for the user to specify this, 1.1 fixes this as you said, so we should be in a better place. We still want to be able to control the speced pods. I'm going to close this, reopen if you're not happy with this.

Longer term we should have something that controls the network configuration (just like Serving), I'd rather create a new issue for it since it's not necessarily limited to the Annotations only.

Also, the ServiceEntry is not sufficient for some protocols because sometimes you need to punch multiple ports through and don't know ahead of time which they are.

**Comment by vaikas-google** [comment]
An example usage is here:
https://github.com/vaikas-google/ftp
